demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -1169,15 +1191,112 @@ export class DemoMeetingApp
       }
       const tileId = videoTile.id();
       const tileIndex = this.tileIdToTileIndex[tileId];
-      this.getStats(tileIndex);
       if (tileState.localTile) {
-        this.statsCollector.showUpstreamStats(tileIndex);
+        this.showVideoUpstreamStats(tileIndex, tileState.boundAttendeeId, videoUpstreamMetricReport);
       } else {
-        this.statsCollector.showDownstreamStats(tileIndex);
+        this.showVideoDownstreamStats(tileIndex, tileState.boundAttendeeId, videoDownstreamMetricReport);
       }
     }
   }
 
+  showVideoUpstreamStats(tileIndex: number, attendeeId: string, videoUpstreamMetricReport: { [id: string]: { [id: string]: {} } }): void {
+    this.showVideoStats(
+      tileIndex,
+      StreamDirection.Upstream,
+      this.videoUpstreamMetricsKeyStatsToShow,
+      videoUpstreamMetricReport[attendeeId]
+    );
+  }
+
+  showVideoDownstreamStats(tileIndex: number, attendeeId: string, videoUpstreamMetricReport: { [id: string]: { [id: string]: {} } }): void {
```

Comment:
those two functions could collapse into one ?

Reply:
also typo `videoUpstreamMetricReport` should be downstream

Reply:
Agreed. I put a comment bellow that we should merge these two observable methods into one also.